bde/ml/models.py

```diff
@@ -901,17 +1137,42 @@ def predict(
         Array
             Predicted labels.
         """
-        bde.utils.utils.check_predict_input(X)
-        chex.assert_equal(self.is_fitted_, True)
-        return jax.pmap(
+        bde.utils.utils.check_predict_input(X, self.is_fitted_)
+        res = jax.lax.scan(
+            f=lambda carry, params: (0.0, self.model_.apply(params, X)),
+            init=0.0,
+            xs=self.samples_,
+        )[1]
+        return res.mean(axis=0)
```

Comment:
this mean prediction should definitely be an option but add the option with a further argument to get predictions by sampling from the fitted Gaussians. The argument could be something like `n_samples` and should indicate how many predictions should be sampled for each parameterized Gaussian (for each posterior sample). In such a way we allow sampling from the empirical posterior predictive distribution. Otherwise we would discard any information about the learned log variance. Same goes for de and credibility intervals

Reply:
By "sampling from the fitted Gaussians", do you mean that after evaluating the `mean` and `std` for each sampled netwrok we should use them to parametrize `n_mcmc_samples` Gaussian distributions, and samples from each of them `n_samples` points (in total `n_samples` $\times$ `n_mcmc_samples` points), and use these points to evaluate the mean prediction and the confidence interval?

Reply:
Yes

Reply:
Ok, I implemented sampling from the predicted distributions.
Now `predict` returns the mean of these samples and `predict_with_credibility_eti` returns (median, low_cred, high_cred).